packages/fast-components-react-base/src/toolbar/toolbar.tsx

```diff
@@ -0,0 +1,355 @@
+        ) {
+            // reset if initialFocusIndex prop has changed
+            if (prevProps.initialFocusIndex !== this.props.initialFocusIndex) {
+                this.setDefaultFocusItem(this.getFocusableItems());
+                return;
+            }
+
+            this.validateCurrentFocusItem();
+        }
+    }
+
+    /**
+     * Ensures that the current focus item path is valid
+     * and resets if not
+     */
+    private validateCurrentFocusItem = (): void => {
+        const focusableItems: HTMLElement[] = this.getFocusableItems();
+
+        if (focusableItems.length === 0) {
+            this.setState({
+                focusItemPath: "-1",
+            });
+            return;
+        }
+
+        if (
+            // reset default focus if the old focused item does not exist
+            this.getItemIndex(focusableItems, this.state.focusItemPath) === -1
+        ) {
+            this.setDefaultFocusItem(focusableItems);
+        }
+    };
+
+    /**
+     * Sets the current focuus item path based on default focus item
+     */
+    private setDefaultFocusItem = (focusableItems: HTMLElement[]): void => {
+        let initialFocusItemPath: number[] = [0];
+
+        if (!isNil(this.props.initialFocusIndex)) {
+            initialFocusItemPath = Array.isArray(this.props.initialFocusIndex)
+                ? this.props.initialFocusIndex
+                : [this.props.initialFocusIndex];
+        }
+
+        let itemPath: string = initialFocusItemPath.toString();
+
+        if (this.getItemIndex(focusableItems, itemPath) === -1) {
+            // default to first focusable widget
+            itemPath = focusableItems[0].getAttribute(Toolbar.toolbarItemAttributeName);
+        }
+
+        this.setState({
+            focusItemPath: itemPath,
+        });
+    };
+
+    /**
+     * Handle the keydown event
+     */
+    private handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>): void => {
+        // increment focused widget on arrow keys
+        if (this.props.orientation === Orientation.horizontal) {
+            switch (e.keyCode) {
+                case keyCodeArrowRight:
+                    e.preventDefault();
+                    this.incrementFocus(this.state.focusItemPath, 1);
+                    break;
+
+                case keyCodeArrowLeft:
+                    e.preventDefault();
+                    this.incrementFocus(this.state.focusItemPath, -1);
+                    break;
+            }
+        } else {
+            switch (e.keyCode) {
+                case keyCodeArrowDown:
+                    e.preventDefault();
+                    this.incrementFocus(this.state.focusItemPath, 1);
+                    break;
+
+                case keyCodeArrowUp:
+                    e.preventDefault();
+                    this.incrementFocus(this.state.focusItemPath, -1);
+                    break;
+            }
+        }
+
+        // shortcuts
```

Comment:
Is there a reason you initialize `focusableWidgets` to null? Both cases assign this to the product  of `this.getFocusableItems()`

Reply:
redundant, removed.  And renamed to focusableItems (at one point I was referring to focusable things as "widgets" but decided "items" was better, this was an artifact).